packages/util/src/object.ts

```diff
@@ -108,24 +110,103 @@ export const assoc = <T>(key: string, value: any, obj: T) => ({
+ * @param key - Name of the header.
+ * @param value - Value of the header.
+ * @returns - An object containing the given key and value of an empty object.
  */
-export const flatten = (input: any[]): any[] => {
-  const flatResult: any[] = [];
-  const stack: any[] = [...input];
+export function toSanitizedObject(
+  key: string,
+  value: any
+): Record<string, any> {
+  return isNullish(key) || isNullish(value) ? {} : { [key]: value };
+}
 
-  while (stack.length > 0) {
-    const current = stack.pop();
-    if (!Array.isArray(current)) {
-      flatResult.push(current);
-    } else {
-      stack.push(...current);
-    }
-  }
+/**
+ * Pick a partial object from the given object indepdendent of casing - if available.
+ * @param obj - Object to pick the given key from.
+ * @param keys - Keys of the pair to be picked.
+ * @returns - An object containing the given key-value pairs in its original case or an empty object if none of them are found.
+ */
+export function pickIgnoreCase<T extends Record<string, any>>(
+  obj: T = {} as T,
+  ...keys: string[]
+): Partial<Pick<T, typeof keys[number]>> {
+  return keys.reduce((filteredHeaders, providedKey) => {
+    const originalKey = Object.keys(obj).find(
+      objKey => objKey.toLowerCase() === providedKey.toLowerCase()
+    );
 
-  return flatResult.reverse();
-};
+    return {
+      ...filteredHeaders,
+      ...(originalKey && { [originalKey]: obj[originalKey] })
+    };
+  }, {});
+}
+
+/**
+ * Pick the value of an object based on the given key, independent of casing.
+ * @param obj - Object to be searched for the given key.
+ * @param key - Key of the value to pick.
+ * @returns The value of for the given key or undefined if not available.
+ */
+export function pickValueIgnoreCase<T extends Record<string, any>>(
+  obj: T = {} as T,
+  key: string
+): any | undefined {
+  return Object.values(pickIgnoreCase(obj, key))[0];
+}
+
+/**
+ * Pick entries that have non-nullish values (and keys).
+ * @param obj - An object to pick frmo.
+ * @returns - A filtered object containing only headers with non-nullish values.
+ */
+export function pickNonNullishIgnoreCase<T extends Record<string, any>>(
+  obj: T = {} as T
+): Partial<T> {
+  return Object.entries(obj)
+    .filter(([key, value]) => !isNullish(key) && !isNullish(value))
+    .reduce((filtered, [key, value]) => ({ ...filtered, [key]: value }), {});
+}
+
+/**
+ * Create an object by merging the `right` object into the `left` object ignoring casing, but keeping the right casing. Only keys present in the left object will be present in the merged object.
+ * @param left - Object to merge into. They keys of this object will be present in the returned object.
+ * @param right - Object to merge. Only keys in `left` will be considered for merging.
+ * @returns - An object containing all keys from the `left` object, where headers present in the `right` object are replaced. Note that the casing used by `right` will be used.
```

Comment:
mention for all merge methods that the resulting object contains shallow copies of the merged objects.

Reply:
Done.